File: app/src/main/java/com/example/java/android1/java_android_notes/dialogs/DialogAddFragment.java
package com.example.java.android1.java_android_notes.dialogs;

import android.app.AlertDialog;
import android.app.Dialog;
import android.os.Bundle;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;

import com.example.java.android1.java_android_notes.R;
import com.example.java.android1.java_android_notes.data.DataNote;
import com.example.java.android1.java_android_notes.data.DataNoteSource;
import com.example.java.android1.java_android_notes.data.DataNoteSourceFirebase;
import com.google.android.material.button.MaterialButton;
import com.google.android.material.textfield.TextInputEditText;

import java.text.SimpleDateFormat;
import java.util.Date;

public class DialogAddFragment extends DialogFragment {

    private static final String KEY_NOTE_TITLE = "DialogAddFragment.Note.title";
    private static final String KEY_NOTE_DESCRIPTION = "DialogAddFragment.Note.title";

    private TextInputEditText mEditTitle;
    private TextInputEditText mEditText;

    public DialogAddFragment() {

    }

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
        View view = requireActivity().getLayoutInflater().inflate(R.layout.fragment_add_note, null, false);

        mEditTitle = view.findViewById(R.id.add_note_title);
        mEditText = view.findViewById(R.id.add_note_text);
        MaterialButton materialButton = view.findViewById(R.id.btn_add_note);
        DataNoteSource dataNoteSource = DataNoteSourceFirebase.getInstance();
        if (savedInstanceState != null) {
            mEditTitle.setText(savedInstanceState.getString(KEY_NOTE_TITLE));
            mEditText.setText(savedInstanceState.getString(KEY_NOTE_DESCRIPTION));
        }

        AlertDialog.Builder builder = new AlertDialog.Builder(requireActivity());
        builder.setTitle("Add Note").setView(view).setCancelable(false);
        SimpleDateFormat format = new SimpleDateFormat("dd.MM.yyyy");
        materialButton.setOnClickListener((v) -> {
            if (!mEditTitle.getText().toString().trim().equals("")) {
                dataNoteSource.createItem(new DataNote(mEditTitle.getText().toString(),
                        mEditText.getText().toString(), false, format.format(new Date())));
                dismiss();
            }
        });
        return builder.create();
    }

    @Override
    public void onSaveInstanceState(@NonNull Bundle outState) {
        super.onSaveInstanceState(outState);
        outState.putString(KEY_NOTE_TITLE, mEditTitle.getText().toString());
        outState.putString(KEY_NOTE_DESCRIPTION, mEditText.getText().toString());
    }
}

File: app/src/main/java/com/example/java/android1/java_android_notes/dialogs/DialogAddFragment.kt
package com.example.java.android1.java_android_notes.dialogs

import android.annotation.SuppressLint
import android.app.AlertDialog
import android.app.Dialog
import android.os.Bundle
import androidx.fragment.app.DialogFragment
import com.example.java.android1.java_android_notes.data.DataNote
import com.example.java.android1.java_android_notes.data.DataNoteSource
import com.example.java.android1.java_android_notes.data.DataNoteSourceFirebase.Companion.instance
import com.example.java.android1.java_android_notes.databinding.FragmentAddNoteBinding
import java.text.SimpleDateFormat
import java.util.*

class DialogAddFragment : DialogFragment() {

    private var _binding: FragmentAddNoteBinding? = null
    private val binding get() = _binding!!

    @SuppressLint("SimpleDateFormat")
    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        _binding = FragmentAddNoteBinding.inflate(requireActivity().layoutInflater)
        val dataNoteSource: DataNoteSource? = instance
        if (savedInstanceState != null) {
            binding.addNoteTitle.setText(savedInstanceState.getString(KEY_NOTE_TITLE))
            binding.addNoteText.setText(savedInstanceState.getString(KEY_NOTE_DESCRIPTION))
        }
        val builder = AlertDialog.Builder(requireActivity())
        builder.setTitle("Add Note").setView(binding.root).setCancelable(false)
        val format = SimpleDateFormat("dd.MM.yyyy")
        binding.btnAddNote.setOnClickListener {
            if (binding.addNoteTitle.text.toString().trim { it <= ' ' } != "") {
                dataNoteSource?.createItem(
                    DataNote(
                        noteName = binding.addNoteTitle.text.toString(),
                        noteDescription = binding.addNoteText.text.toString(),
                        noteFavorite = false,
                        noteDate = format.format(Date())
                    )
                )
                dismiss()
            }
        }
        return builder.create()
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        outState.putString(KEY_NOTE_TITLE, binding.addNoteTitle.text.toString())
        outState.putString(KEY_NOTE_DESCRIPTION, binding.addNoteText.text.toString())
    }

    companion object {
        private const val KEY_NOTE_TITLE = "DialogAddFragment.Note.title"
        private const val KEY_NOTE_DESCRIPTION = "DialogAddFragment.Note.title"
    }
}